tests/sample.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * SLA bridge and channel creation tests.
+ *
+ */
+"use strict";
+
+/*global describe:false*/
+/*global afterEach:false*/
+
+var util = require('util');
+var Emitter = require('events').EventEmitter;
+
+var mockClient;
+var mockChannel;
+var mockBridge;
+// The channels in existance
+var channels = [];
+// The bridges in existance (should only be one)
+var bridges = [];
+// The channels in the bridges
+var bridge_channels = [];
+// Tests if a bridge has been created or not on a particular test. Resets
+var bridge_created = 0;
+// Mock value for whether the endpoint is valid or not
+var valid = 1;
+// Conditional for whether or not StasisStart has been passed yet
+var pastStasis = 0;
+
+// The mocked up version of the callback error function
+var end= function(err){console.log(err);}
+
+// Millesecond delay for mock requests
+var asyncDelay = 100;
+
+/**
+ * Returns a mock client that houses the bridges and can create both bridges
+ * and channels.
+ */
+
+var getMockClient = function() {
+  if(mockClient) {
+    return mockClient;
+  }
+  var Client = function() {
+    this.bridges = {
+      list: function(cb) {
+        console.log(bridges);
+        cb(null, bridges);
+        return bridges;
+      },
+      create: function(param, cb) {
+        bridge_created = 1;
+        console.log(param);
+        bridges.push(getMockBridge(param));
+        console.log(bridges)
+        cb(null, bridges[0]);
+      }
+    };
+    this.Channel= function() {
+      return getMockChannel();
+      }
+  };
+  mockClient = new Client();
+  return mockClient;
+};
+
+/**
+ * Returns a mock bridge that has a bridge_type and id, and can add channels
+ * to its array of channels. Emits a ChannelEnteredBridge event
+ */
+
+var getMockBridge = function(param) {
+  var Bridge = function(param) {
+    console.log(param.type);
+    this.bridge_type = param.type;
+    this.id = "1";
+    this.addChannel= function(input, cb) {
+      console.log("TREX");
+      pastStasis = 1;
+      var channel = channels.filter(function(testChan) {
+        return testChan.id === '1';
+      })[0];
+      bridge_channels.push(channel);
+      cb(null);
+      console.log(channel);
+      this.emit('ChannelEnteredBridge', {channel: {id: channel.id}});
+    };
+  };
+  util.inherits(Bridge, Emitter);
+  mockBridge = new Bridge(param);
+  return mockBridge;
+};
+
+/**
+ * Returns a mock channel that has an id, and can use originate to add a channel
+ * to the array of total channels. Emits a StasisStart event when originating.
+ */
+
+var getMockChannel = function() {
+  var Channel = function() {
+    this.id = '1';
+    this.originate = function(input, cb) {
+      console.log(input);
+      var self = this;
+      setTimeout(function() {
+        if(valid == 1) {
+          console.log(self);
+          self.emit('StasisStart', self, self);
```

Comment:
The first object passed to the event should not simply be a channel but rather an event object that includes a channel key with data about the channel. If a library you are testing looks at event.channel.id for example it would fail.